manager/controllers/app/select_modules.go

```diff
@@ -85,9 +85,20 @@ func (p *PlotterGenerator) FindPaths(item *DataInfo, appContext *app.FybrikAppli
 	}
 	// this node is either a virtual endpoint, or a datastore
 	NodeFromAppRequirements := Node{Connection: &item.Context.Requirements.Interface}
+	// for write flow the dataformat in the application is not needed
+	WriteNodeFromAppRequirements := Node{
+		Connection: &taxonomy.Interface{
+			Protocol: item.Context.Requirements.Interface.Protocol,
+		},
+	}
 
 	if item.Context.Flow == taxonomy.WriteFlow {
-		source = NodeFromAppRequirements
+		if item.Context.Requirements.Interface.DataFormat != "" && item.Context.Requirements.Interface.DataFormat != item.DataDetails.Details.DataFormat {
```

Comment:
Not sure this should result in an error.  There may be modules available that can transform between the formats.  If so, they should be included in the data path.  If there are no such modules available then an error should be returned